### PR TITLE
E2E test script: Extend supported providers

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Towards ##3256

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: api/hack/ci/ci-run-minimal-conformance-tests.sh [hunk 1]
```diff
@@ -40,18 +41,37 @@ function cleanup {
...
       kubectl describe cluster -l worker-name=$BUILD_ID|egrep -vi 'APIKey|ProjectID'
-    elif [[ $provider = "gcp" ]]; then
+    elif [[ $provider == "gcp" ]]; then
       kubectl describe cluster -l worker-name=$BUILD_ID|egrep -vi 'Service Account'
+    elif [[ $provider == "azure" ]]; then
+      kubectl describe cluster -l worker-name=$BUILD_ID|egrep -vi 'ClientID|ClientSecret|SubscriptionID|TenantID'
+    elif [[ $provider == "digitalocean" ]]; then
+      kubectl describe cluster -l worker-name=$BUILD_ID|egrep -vi 'Token'
+    elif [[ $provider == "hetzner" ]]; then
+      kubectl describe cluster -l worker-name=$BUILD_ID|egrep -vi 'Token'
```
Comment (nikhita) [inline api/hack/ci/ci-run-minimal-conformance-tests.sh:56]: I think this needs to be `API Token` for digital ocean and hetzner

https://github.com/kubermatic/kubermatic/blob/dcb8dcc8661e0a88b90c58dafdb999e3ec5bcca1/api/cmd/conformance-tests/main.go#L166-L167

Comment (alvaroaleman) [reply api/hack/ci/ci-run-minimal-conformance-tests.sh:56]: Actually no, this is right the way it is:

DO:
```
k describe cluster 5g7zrr4vsn|cut -d ':' -f1|grep -C3 Tok
  Cloud
    Dc
    Digitalocean
      Token
  Cluster Network
    Dns Domain
    Pods
```

Hetzner:
```
$ k describe cluster qgdp8gdmwm|grep -i token -C3|cut -d: -f1
  Cloud
    Dc
    Hetzner
      Token
  Cluster Network
    Dns Domain
    Pods
```

**Review by nikhita — APPROVED** [review]
/lgtm
/approve

**Comment by alvaroaleman** [comment]
Why did you `/hold` this?

**Comment by nikhita** [comment]
Added the hold thinking the related tests/jobs were optional presubmits...but later realized they are periodics. :woman_facepalming: 

Please cancel the hold at your discretion. :)

**Pull request merged**